### DISCUSSION TITLE
some problems about BackgroundVelocityField change with time

### DISCUSSION THREAD

**Discussion by Tinydog8** [CliMA/Oceananigans.jl#2793] [Computational science!] [answered]
Hello,
I'm trying to create a simulation with a background velocity field which will change with time(for example,ubg=sin(t)).I'm not sure that if  ”BackgroundField“ takes the effect of non inertial reference system into account？or maybe I should use ”Forcing“ function in this situation？
Thanks in advance!

**Comment by glwagner** [comment, marked as answer]
`BackgroundField` doesn't change the frame of reference; it changes the meaning of the prognostic variables. I think `U = sin(t)` should be fine! But check out the equations in the docs to see if it's really the model you're looking for:

https://clima.github.io/OceananigansDocumentation/stable/physics/nonhydrostatic_model/

**Reply by Tinydog8** [reply]
> `BackgroundField` doesn't change the frame of reference; it changes the meaning of the prognostic variables. I think `U = sin(t)` should be fine! But check out the equations in the docs to see if it's really the model you're looking for:
> 
> https://clima.github.io/OceananigansDocumentation/stable/physics/nonhydrostatic_model/

Thank you very much! If I defined the background velocity ubg changed with time, then the total velocity can be divided into u=v+ubg, v
  is the resolved velocity. Take u into momentum equation, because the ubg change with time, there should have an additional term ∂(ubg)/∂t. But it seems not be mentioned in the document of "nonhydrostatic model", or maybe it is included in other terms? Thanks again！

**Reply by glwagner** [reply]
For sure, including a background field is _not_ equivalent to simply substituting `u = u_bg + u'` into the momentum equations, and solving for `u'`.

I don't think you want to include the term `∂(ubg)/∂t`. Consider: if you include that term, then the solution for the perturbation velocity is simply `u = - u_bg`, and you are back where you started.

Including a background field is only useful when we _omit_ the "pure" background terms, and only include terms associated with background-perturbation interaction (right now we only include nonlinear advection terms and not _potential_ nonlinear diffusion terms that would be associated with LES. This is a limitation.) One of the most important uses of `BackgroundField` is when we want to simulate periodic perturbations around a non-periodic flow (like Eady turbulence) in a periodic domain.

If you want dynamics that include both background dynamics _and_ perturbation dynamics, then you should set your simulation up with forcing.

 The docs just write down the equations that are solved when you use the background velocity. They don't explain _why_ you want to solve that equation --- perhaps that's something we should add.

**Reply by Tinydog8** [reply]
Thanks a lot!！！！
